src/components/Search/Search.js

```diff
@@ -221,7 +221,7 @@ Search.propTypes = {
    * Handler that runs when an item from the search filter dropdown is selected
    * Returns the selected filter value.
    */
-  onFilterSelect: PropTypes.func,
+  onFilterSelect: PropTypes.func.isRequired,
```

Comment:
`onFilterSelect` is only required if `filters` is provided. I think I tried to write a dependant type thing but it got messy[0] and so fell back to the run-time hack in the defaultProps below.

I think it would be nice to have onFilterSelect optional somehow, including https://github.com/jamiebuilds/react-required-if could be an okay solution, or making our own similar wrapper and chucking it in utils could work too! WDYT?

[0] https://stackoverflow.com/questions/42299335/react-props-set-isrequired-on-a-prop-if-another-prop-is-null-empty

Reply:
Yep sounds good. Added `react-required-if` as no point re-implementing that 👌